From 73fa064fcd4b4b37fccf05effc29cff431ca8091 Mon Sep 17 00:00:00 2001
From: Wulf C. Krueger <philantrop@exherbo.org>
Date: Tue, 6 Jan 2009 11:45:12 +0100
Subject: [PATCH] Add the series_index back to the comments.

---
 src/calibre/library/cli.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/calibre/library/cli.py b/src/calibre/library/cli.py
index efb6af7..80371fd 100644
--- a/src/calibre/library/cli.py
+++ b/src/calibre/library/cli.py
@@ -83,7 +83,7 @@ STANZA_TEMPLATE='''\
       <link py:if="record['cover']" rel="x-stanza-cover-image-thumbnail" type="image/png" href="${quote(record['cover'].replace(sep, '/')).replace('http%3A', 'http:')}" />
       <content type="xhtml">
           <div xmlns="http://www.w3.org/1999/xhtml">
-              <py:for each="f in ('authors','publisher','rating','tags','series', 'isbn')">
+              <py:for each="f in ('authors','publisher','rating','tags','series','series_index','isbn')">
               <py:if test="record[f]">
               ${f.capitalize()}:${unicode(', '.join(record[f]) if f=='tags' else record[f])}<br/>
               </py:if>
-- 
1.6.1

