From 17cd5c23b3ff0353e79c76813fb5372ce8215bad Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Sat, 12 Sep 2009 14:25:05 -0600
Subject: [PATCH 1/3] IGN: Make --root a synonym for --prefix for the develop and install commands

---
 setup/install.py |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup/install.py b/setup/install.py
index 63c288a..16091f2 100644
--- a/setup/install.py
+++ b/setup/install.py
@@ -43,7 +43,7 @@ class Develop(Command):
     sub_commands = ['build', 'resources', 'gui']
 
     def add_options(self, parser):
-        parser.add_option('--prefix',
+        parser.add_option('--prefix', '--root',
             help='Binaries will be installed in <prefix>/bin')
 
     def pre_sub_commands(self, opts):
@@ -137,7 +137,7 @@ class Install(Develop):
     sub_commands = ['build', 'gui']
 
     def add_options(self, parser):
-        parser.add_option('--prefix', help='Installation prefix')
+        parser.add_option('--prefix', '--root', help='Installation prefix')
         parser.add_option('--libdir', help='Where to put calibre library files')
         parser.add_option('--bindir', help='Where to install calibre binaries')
         parser.add_option('--sharedir', help='Where to install calibre data files')
-- 
1.6.4.2

