From 311e40a4a8bb912a712bcf5adcc4ad110bebd360 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Sat, 12 Sep 2009 14:27:59 -0600
Subject: [PATCH 2/3] IGN:develop and install commands will now create --bindir if it does not already exist

---
 setup/install.py |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/setup/install.py b/setup/install.py
index 16091f2..76ac277 100644
--- a/setup/install.py
+++ b/setup/install.py
@@ -119,6 +119,8 @@ class Develop(Command):
                 path=self.path, resources=self.resources,
                 extensions=self.extensions)
         path = self.j(self.bindir, name)
+        if not os.path.exists(self.bindir):
+            os.makedirs(self.bindir)
         self.info('Installing binary:', path)
         open(path, 'wb').write(script)
         os.chmod(path, self.MODE)
-- 
1.6.4.2

