Source: philantrop@exherbo.org
Upstream: No.
Reason: Fix slsh.

diff -Naur slsh.orig/Makefile.in slsh/Makefile.in
--- slsh.orig/Makefile.in	2008-09-27 11:28:31.000000000 +0200
+++ slsh/Makefile.in	2008-09-27 11:34:50.000000000 +0200
@@ -72,7 +72,7 @@
 #
 all: slsh_exe
 slsh_exe: slsh.o readline.o
-	$(CC) $(CFLAGS) slsh.o readline.o -o slsh_exe $(LDFLAGS) $(SRC_LIBS)
+	$(CC) $(CFLAGS) slsh.o readline.o -o slsh_exe $(LDFLAGS) -lncurses $(SRC_LIBS)
 slsh: slsh.o readline.o
 	$(CC) $(CFLAGS) slsh.o readline.o -o slsh $(LDFLAGS) -lncurses $(SRC_LIBS)
 slsh.o: slsh.c slsh.h config.h Makefile
