From e32bf509572ef84f0a89d44326daaf2c6f92e0eb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kim=20H=C3=B8jgaard-Hansen?= <kimrhh@exherbo.org>
Date: Thu, 19 Nov 2009 13:47:39 +0100
Subject: [PATCH] remove test failure submits via network

---
 Tests/CMakeLists.txt |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Tests/CMakeLists.txt b/Tests/CMakeLists.txt
index 6fbe57d..7df63c1 100644
--- a/Tests/CMakeLists.txt
+++ b/Tests/CMakeLists.txt
@@ -1255,7 +1255,7 @@ ${CMake_BINARY_DIR}/bin/cmake -DVERSION=CVS -P ${CMake_SOURCE_DIR}/Utilities/Rel
 
   set(ctest_coverage_labels_args "")
 
-  foreach(drop_method cp ftp http https scp xmlrpc)
+  foreach(drop_method cp)
     # Cycle through these values each time through the loop:
     if(ctest_coverage_labels_args STREQUAL "")
       set(ctest_coverage_labels_args "LABELS Everything")
-- 
1.6.5.2

