diff -ur eselect-1.0.10.orig/libs/package-manager.bash.in eselect-1.0.10/libs/package-manager.bash.in
--- eselect-1.0.10.orig/libs/package-manager.bash.in	2007-06-17 00:15:04.000000000 +0200
+++ eselect-1.0.10/libs/package-manager.bash.in	2007-09-24 20:09:40.000000000 +0200
@@ -120,6 +120,20 @@
     esac
 }
 
+get_config_protect_dirs() {
+    local manager=$(package-manager)
+    case ${manager} in
+        paludis)
+            cat "$(${PALUDIS_COMMAND:-@PALUDIS@} --configuration-variable installed location)"/.cache/all_CONFIG_PROTECT
+            ;;
+        portage)
+            envvar sys-devel/gcc CONFIG_PROTECT
+            ;;
+        *)
+            die "Unknown package manager: \"${manager}\"!"
+    esac
+}
+
 # package-manager PRIVATE
 # Return the package manager we're going to use.
 package-manager() {
@@ -142,5 +156,3 @@
 }
 
 # vim: set sw=4 et sts=4 tw=80 :
-
-
diff -ur eselect-1.0.10.orig/modules/config.eselect eselect-1.0.10/modules/config.eselect
--- eselect-1.0.10.orig/modules/config.eselect	2007-06-17 00:15:06.000000000 +0200
+++ eselect-1.0.10/modules/config.eselect	2007-09-24 20:44:16.000000000 +0200
@@ -2,7 +2,7 @@
 # Distributed under the terms of the GNU General Public License v2
 # $Id: config.eselect 370 2007-06-16 22:18:30Z kugelfang $
 
-inherit portage
+inherit package-manager
 
 DESCRIPTION="Handle configuration file merges"
 MAINTAINER="ciaranm@gentoo.org"
@@ -37,7 +37,7 @@
 
 config_protect_dirs() {
 	if [[ -z ${CONFIG_UPDATE_DIRS} ]] ; then
-		export CONFIG_UPDATE_DIRS=$(envvar sys-devel/gcc CONFIG_PROTECT )
+		export CONFIG_UPDATE_DIRS=$(get_config_protect_dirs )
 	fi
 	echo ${CONFIG_UPDATE_DIRS}
 }
@@ -414,7 +414,7 @@
 								break
 								;;
 							m|me|mer|merg|merge)
-								do_action config reject ${command}
+								do_action config merge ${command}
 								show_list=yes
 								break
 								;;
diff -ur eselect-1.0.10.orig/modules/Makefile.in eselect-1.0.10/modules/Makefile.in
--- eselect-1.0.10.orig/modules/Makefile.in	2007-06-17 00:17:48.000000000 +0200
+++ eselect-1.0.10/modules/Makefile.in	2007-09-24 18:15:25.000000000 +0200
@@ -130,6 +130,7 @@
 safe_scripts = \
 		 bashcomp.eselect \
 		 binutils.eselect \
+		 config.eselect \
 		 env.eselect \
 		 kernel.eselect \
 		 mailer.eselect \
@@ -137,7 +138,6 @@
 		 rc.eselect
 
 dodgy_scripts = \
-		 config.eselect \
 		 cow.eselect
 
 @DODGY_MODULES_FALSE@modules_DATA = $(safe_scripts)
