From b2cfac8ace230d3cfafd7f3c15d5610bc2e6e044 Mon Sep 17 00:00:00 2001
From: Petr Machata <pmachata@redhat.com>
Date: Fri, 30 Nov 2012 11:30:05 +0100
Subject: [PATCH] Linux VDSO can be also named linux-vdso{32,64}.so.1

---
 sysdeps/linux-gnu/proc.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/sysdeps/linux-gnu/proc.c b/sysdeps/linux-gnu/proc.c
index 6aedde5..5adfb16 100644
--- a/sysdeps/linux-gnu/proc.c
+++ b/sysdeps/linux-gnu/proc.c
@@ -521,7 +521,9 @@ crawl_linkmap(struct Process *proc, struct lt_r_debug_64 *dbg)
 		 * kludge is about the best that we can do.  */
 		if (*lib_name == 0
 		    || strcmp(lib_name, "linux-vdso.so.1") == 0
-		    || strcmp(lib_name, "linux-gate.so.1") == 0)
+		    || strcmp(lib_name, "linux-gate.so.1") == 0
+		    || strcmp(lib_name, "linux-vdso32.so.1") == 0
+		    || strcmp(lib_name, "linux-vdso64.so.1") == 0)
 			continue;
 
 		/* Do we have that library already?  */
-- 
1.7.2.5

