diff --git a/ext/syck/emitter.c b/ext/syck/emitter.c
index 9c8ab8d..0a69fb2 100644
--- a/ext/syck/emitter.c
+++ b/ext/syck/emitter.c
@@ -554,7 +554,7 @@ syck_scan_scalar( int req_width, char *cursor, long len )
          cursor[0] == '>' || cursor[0] == '\'' ||
          cursor[0] == '"' || cursor[0] == '#' ||
          cursor[0] == '%' || cursor[0] == '@' ||
-         cursor[0] == '&' ) {
+         cursor[0] == '&' || cursor[0] == '`') {
             flags |= SCAN_INDIC_S;
     }
     if ( ( cursor[0] == '-' || cursor[0] == ':' ||
