Upstream: No

From 3a672ac92795d35bcbf19d42bb4f8081b54e8521 Mon Sep 17 00:00:00 2001
From: Ingmar Vanhassel <ingmar@exherbo.org>
Date: Sat, 11 Oct 2008 17:45:02 +0200
Subject: [PATCH] Use a relative path for this test, since opening /none/such as non-root will generate permission errors on most systems.

---
 doc/m4.texinfo |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/doc/m4.texinfo b/doc/m4.texinfo
index 940bd9c..768ea8e 100644
--- a/doc/m4.texinfo
+++ b/doc/m4.texinfo
@@ -6838,12 +6838,12 @@
 
 @c Detect inability to freeze.
 
-@comment options: -F /none/such
+@comment options: -F ./none/such
 @comment status: 1
 @example
-$ @kbd{m4 -F /none/such}
+$ @kbd{m4 -F ./none/such}
 ^D
-@error{}m4: cannot open `/none/such': No such file or directory
+@error{}m4: cannot open `./none/such': No such file or directory
 @end example
 @end ignore
 
-- 
1.6.0.2

