From bc791b281ac78c7ea99bff5ce7d39be5ba28aabe Mon Sep 17 00:00:00 2001
From: troy d. straszheim <troy@resophonic.com>
Date: Wed, 16 Sep 2009 06:04:41 -0700
Subject: [PATCH] don't try to build inspect (where did this guy's CMakeLists.txt go?)

---
 tools/CMakeLists.txt |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index b87c782..aa53f0c 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -86,4 +86,4 @@ endmacro(boost_tool_dependencies)
 add_subdirectory(quickbook)
 add_subdirectory(wave)
 add_subdirectory(bcp)
-add_subdirectory(inspect)
\ No newline at end of file
+# add_subdirectory(inspect)
-- 
1.6.4.3
