Upstream: yes, in branch 2.32
From 9ab02c1428385c30b13e24872960474d44b6866b Mon Sep 17 00:00:00 2001
From: Rafał Mużyło <galtgendo@o2.pl>
Date: Sun, 15 Apr 2012 15:17:07 +0000
Subject: configure: Fix typo in ELF check

Signed-off-by: Colin Walters <walters@verbum.org>
---
diff --git a/configure.ac b/configure.ac
index 849b262..9869daa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1826,7 +1826,7 @@ if test $have_libelf = maybe; then
 
   if test $have_libelf != no; then
     LIBELF_LIBS=-lelf
-    have_libelf = yes
+    have_libelf=yes
   fi
 fi
 
--
cgit v0.9.0.2
