From c04179fe94aa0b1af8c2a7c62111fe81561634d8 Mon Sep 17 00:00:00 2001
From: Ingmar Vanhassel <ingmar@exherbo.org>
Date: Thu, 19 Nov 2009 20:58:33 +0100
Subject: [PATCH] Disable a test that hangs with sydbox

---
 tests/tail-2/inotify-race |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/tests/tail-2/inotify-race b/tests/tail-2/inotify-race
index 0ee2df5..9708909 100755
--- a/tests/tail-2/inotify-race
+++ b/tests/tail-2/inotify-race
@@ -27,6 +27,8 @@ fi
 
 . $srcdir/test-lib.sh
 
+skip_test_ 'This test (using gdb) hangs under sydbox'
+
 # Don't run this test by default because sometimes it's skipped as noted below.
 # Also gdb has a bug in Debian's gdb-6.8-3 at least that causes it to not
 # cleanup and exit correctly when it receives a SIGTERM, thus hanging the test.
-- 
1.6.6.rc2.275.g51e2d

