From 781b07b355da3614aca4225eb7db7ba3018541aa Mon Sep 17 00:00:00 2001
From: Ingmar Vanhassel <ingmar@exherbo.org>
Date: Thu, 19 Nov 2009 20:58:33 +0100
Subject: [PATCH] Disable a test that hangs with sydbox

---
 tests/tail-2/inotify-race |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/tests/tail-2/inotify-race b/tests/tail-2/inotify-race
index b1e1a7c..6533a1b 100755
--- a/tests/tail-2/inotify-race
+++ b/tests/tail-2/inotify-race
@@ -27,6 +27,7 @@ fi
 
 . $srcdir/test-lib.sh
 
+skip_test_ 'This test (using gdb) hangs under sydbox'
 
 touch file || framework_failure
 touch tail.out || framework_failure
-- 
1.6.5.2.433.g23cdb

