diff -urN net-tools-1.60_orig/hostname.c net-tools-1.60/hostname.c
--- net-tools-1.60_orig/hostname.c	2007-07-07 03:40:23.000000000 +0200
+++ net-tools-1.60/hostname.c	2007-07-07 03:47:00.000000000 +0200
@@ -78,6 +78,7 @@
             fprintf(stderr, _("%s: name too long\n"), program_name);
             break;
         default:
+		;
         }
 	exit(1);
     }
@@ -98,6 +99,7 @@
 	    fprintf(stderr, _("%s: name too long\n"), program_name);
 	    break;
 	default:
+		;
 	}
 	exit(1);
     };
@@ -117,6 +119,7 @@
 	    fprintf(stderr, _("%s: name too long\n"), program_name);
 	    break;
 	default:
+		;
 	}
 	exit(1);
     };
@@ -174,6 +177,7 @@
 	printf("%s\n", hp->h_name);
 	break;
     default:
+    	;
     }
 }
 
diff -urN net-tools-1.60_orig/lib/inet_sr.c net-tools-1.60/lib/inet_sr.c
--- net-tools-1.60_orig/lib/inet_sr.c	2007-07-07 03:40:23.000000000 +0200
+++ net-tools-1.60/lib/inet_sr.c	2007-07-07 03:40:44.000000000 +0200
@@ -105,6 +105,7 @@
     case 2:
        isnet = 0; break;
     default:
+    	;
     }
 
     /* Fill in the other fields. */
