From c24bbd821ca4a0fba6e275474b0d10785bb94ee8 Mon Sep 17 00:00:00 2001
From: Elias Pipping <pipping@exherbo.org>
Date: Sat, 30 Jul 2011 19:54:24 +0200
Subject: [PATCH] Respect variables set by configure

---
 Makefile.config.in |   29 ++++++++++++++++++++++-------
 1 files changed, 22 insertions(+), 7 deletions(-)

diff --git a/Makefile.config.in b/Makefile.config.in
index 9bdb075..7950c9f 100644
--- a/Makefile.config.in
+++ b/Makefile.config.in
@@ -53,6 +53,23 @@ PROJ_SRC_ROOT   := $(LLVM_SRC_ROOT)
 PROJ_SRC_DIR    := $(call realpath, $(LLVM_SRC_ROOT)/$(patsubst $(PROJ_OBJ_ROOT)%,%,$(PROJ_OBJ_DIR)))
 prefix          := @prefix@
 PROJ_prefix     := $(prefix)
+exec_prefix     := @exec_prefix@
+bindir          := @bindir@
+PROJ_bindir     := $(bindir)
+libdir          := @libdir@
+PROJ_libdir     := $(libdir)
+datadir         := @datadir@
+PROJ_datadir    := $(datadir)
+docsdir         := @docdir@
+PROJ_docsdir    := $(docsdir)
+etcdir          := @sysconfdir@/llvm
+PROJ_etcdir     := $(etcdir)
+includedir      := @includedir@
+PROJ_includedir := $(includedir)
+infodir         := @infodir@
+PROJ_infodir    := $(infodir)
+mandir          := @mandir@
+PROJ_mandir     := $(mandir)
 PROJ_VERSION    := $(LLVMVersion)
 else
 ifndef PROJ_SRC_ROOT
@@ -73,13 +89,6 @@ endif
 PROJ_SRC_DIR := $(call realpath, $(PROJ_SRC_ROOT)/$(patsubst $(PROJ_OBJ_ROOT)%,%,$(PROJ_OBJ_DIR)))
 prefix          := $(PROJ_INSTALL_ROOT)
 PROJ_prefix     := $(prefix)
-ifndef PROJ_VERSION
-PROJ_VERSION := 1.0
-endif
-endif
-
-LLVMMAKE := $(LLVM_SRC_ROOT)/make
-
 PROJ_bindir     := $(PROJ_prefix)/bin
 PROJ_libdir     := $(PROJ_prefix)/lib
 PROJ_datadir    := $(PROJ_prefix)/share
@@ -88,6 +97,12 @@ PROJ_etcdir     := $(PROJ_prefix)/etc/llvm
 PROJ_includedir := $(PROJ_prefix)/include
 PROJ_infodir    := $(PROJ_prefix)/info
 PROJ_mandir     := $(PROJ_prefix)/share/man
+ifndef PROJ_VERSION
+PROJ_VERSION := 1.0
+endif
+endif
+
+LLVMMAKE := $(LLVM_SRC_ROOT)/make
 
 # Determine if we're on a unix type operating system
 LLVM_ON_UNIX:=@LLVM_ON_UNIX@
-- 
1.7.6

