From: Daniel Mierswa <impulze@impulze.org>
Reason: Add Exherbo libstdc++ include paths
Upstream: Unknown

--- ./tools/clang/lib/Frontend/InitHeaderSearch.cpp	2009-10-16 15:42:11.000000000 +0200
+++ ./tools/clang/lib/Frontend/InitHeaderSearch.cpp	2009-10-16 15:51:30.000000000 +0200
@@ -392,6 +392,16 @@
              "i686-pc-linux-gnu",
              "i686-pc-linux-gnu",
              triple);
+
+          // Exherbo 2009-10-16
+          AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.4.1",
+                                      "x86_64-pc-linux-gnu",
+                                      "x86_64-pc-linux-gnu",
+                                      triple);
+          AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.4.1",
+                                      "i686-pc-linux-gnu",
+                                      "i686-pc-linux-gnu",
+                                      triple);
           break;
         case llvm::Triple::FreeBSD:
           // DragonFly
