Source: Written by alip
Upstream: no
Reason: Exherbo rocks!
--- bugz.py	2009-02-19 17:45:51.000000000 +0000
+++ bugz.py	2009-02-19 17:46:02.000000000 +0000
@@ -1168,7 +1168,7 @@
 class PrettyBugz(Bugz):
     options = {
         'base': make_option('-b', '--base', type='string',
-                            default = 'http://bugs.gentoo.org/',
+                            default = 'https://bugs.exherbo.org/',
                             help = 'Base URL of Bugzilla'),
         'user': make_option('-u', '--user', type='string',
                             help = 'Username for commands requiring authentication'),
