From 1a5aea7d3fa6085ab1168b3dcc16749f9d85d1e9 Mon Sep 17 00:00:00 2001
From: Ingmar Vanhassel <ingmar@exherbo.org>
Date: Thu, 2 Jul 2009 10:26:06 +0200
Subject: [PATCH] Exherbo rocks!

---
 bugz/cli.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/bugz/cli.py b/bugz/cli.py
index c9f5311..1e90a50 100644
--- a/bugz/cli.py
+++ b/bugz/cli.py
@@ -137,7 +137,7 @@ def modify_opt_invalid(opt, opt_str, val, parser):
 class PrettyBugz(Bugz):
 	options = {
 		'base': make_option('-b', '--base', type='string',
-							default = 'https://bugs.gentoo.org/',
+							default = 'https://bugs.exherbo.org/',
 							help = 'Base URL of Bugzilla'),
 		'user': make_option('-u', '--user', type='string',
 							help = 'Username for commands requiring authentication'),
-- 
1.6.3.3

