Source: Debian
Upstream: Unknown
Reason: open_nonarchive.test fails

--- a/regress/open_new_but_exists.test
+++ b/regress/open_new_but_exists.test
@@ -3,4 +3,4 @@ program tryopen
 args -e test.zip
 return 1
 file test.zip test.zip test.zip
-stdout opening `test.zip' returned error 10/0
+stdout opening `test.zip' returned error 10
--- a/regress/open_nonarchive.test
+++ b/regress/open_nonarchive.test
@@ -3,4 +3,4 @@ program tryopen
 file Makefile.am Makefile.am Makefile.am
 args Makefile.am
 return 1
-stdout opening `Makefile.am' returned error 19/2
+stdout opening `Makefile.am' returned error 19
--- a/regress/tryopen.c
+++ b/regress/tryopen.c
@@ -90,7 +90,9 @@ main(int argc, char *argv[])
 	return 0;
     }
 
-    printf("opening `%s' returned error %d/%d\n",
-	   fname, ze, errno);
+    printf("opening `%s' returned error %d", fname, ze);
+    if (zip_error_get_sys_type(ze) == ZIP_ET_SYS)
+	printf("/%d", errno);
+    printf("\n");
     return 1;
 }
