From 6ba6a3cd06cdfc0a0eae3bc0daf3ec0178781e40 Mon Sep 17 00:00:00 2001
From: Wulf C. Krueger <philantrop@exherbo.org>
Date: Tue, 28 Sep 2010 13:13:17 +0200
Subject: [PATCH 1/2] Add Exherbo distribution

---
 configure.ac            |   11 ++++++++++-
 units/getty@.service.m4 |    1 +
 2 files changed, 11 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 12cdb0c..7984f3f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -264,7 +264,7 @@ AM_CONDITIONAL(HAVE_XSLTPROC, test x"$XSLTPROC" != x)
 
 AC_PATH_PROG([M4], [m4])
 
-AC_ARG_WITH(distro, AS_HELP_STRING([--with-distro=DISTRO],[Specify the distribution to target: One of fedora, suse, debian, ubuntu, arch, gentoo, slackware or other]))
+AC_ARG_WITH(distro, AS_HELP_STRING([--with-distro=DISTRO],[Specify the distribution to target: One of fedora, suse, debian, ubuntu, arch, gentoo, slackware, exherbo or other]))
 if test "z$with_distro" = "z"; then
         if test "$cross_compiling" = yes; then
                 AC_MSG_WARN([Target distribution cannot be reliably detected when cross-compiling. You should specify it with --with-distro (see $0 --help for recognized distros)])
@@ -278,6 +278,7 @@ if test "z$with_distro" = "z"; then
                 AC_CHECK_FILE(/etc/arch-release,with_distro="arch")
                 AC_CHECK_FILE(/etc/gentoo-release,with_distro="gentoo")
                 AC_CHECK_FILE(/etc/slackware-version,with_distro="slackware")
+                AC_CHECK_FILE(/etc/exherbo-release,with_distro="exherbo")
         fi
         if test "z$with_distro" = "z"; then
                 with_distro=`uname -s`
@@ -350,6 +351,13 @@ case $with_distro in
                 AC_DEFINE(TARGET_SLACKWARE, [], [Target is Slackware])
                 M4_DISTRO_FLAG=-DTARGET_SLACKWARE=1
                 ;;
+        exherbo)
+                SYSTEM_SYSVINIT_PATH=
+                SYSTEM_SYSVRCND_PATH=
+                SPECIAL_SYSLOG_SERVICE=syslog.service
+                AC_DEFINE(TARGET_EXHERBO, [], [Target is Exherbo])
+                M4_DISTRO_FLAG=-DTARGET_EXHERBO=1
+                ;;
         other)
                 AS_IF([test "x$with_syslog_service" = "x"],
                         [AC_MSG_ERROR([With --distro=other, you must pass --with-syslog-service= to configure])])
@@ -399,6 +407,7 @@ AM_CONDITIONAL(TARGET_DEBIAN_OR_UBUNTU, test x"$with_distro" = xdebian -o x"$wit
 AM_CONDITIONAL(TARGET_ARCH, test x"$with_distro" = xarch)
 AM_CONDITIONAL(TARGET_GENTOO, test x"$with_distro" = xgentoo)
 AM_CONDITIONAL(TARGET_SLACKWARE, test x"$with_distro" = xslackware)
+AM_CONDITIONAL(TARGET_EXHERBO, test x"$with_distro" = xexherbo)
 
 AC_DEFINE_UNQUOTED(SPECIAL_SYSLOG_SERVICE, ["$SPECIAL_SYSLOG_SERVICE"], [Syslog service name])
 
diff --git a/units/getty@.service.m4 b/units/getty@.service.m4
index 24247be..76e4c1f 100644
--- a/units/getty@.service.m4
+++ b/units/getty@.service.m4
@@ -11,6 +11,7 @@ m4_ifdef(`TARGET_GENTOO', `m4_define(`GETTY', `/sbin/agetty')')m4_dnl
 m4_ifdef(`TARGET_ARCH', `m4_define(`GETTY', `/sbin/agetty')')m4_dnl
 m4_ifdef(`TARGET_DEBIAN', `m4_define(`GETTY', `/sbin/getty')')m4_dnl
 m4_ifdef(`TARGET_UBUNTU', `m4_define(`GETTY', `/sbin/getty')')m4_dnl
+m4_ifdef(`TARGET_EXHERBO', `m4_define(`GETTY', `/sbin/agetty')')m4_dnl
 m4_dnl
 [Unit]
 Description=Getty on %I
-- 
1.7.3.2

