From dc21c46b749aa9ba74a311a248ac89084d392fcd Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Mon, 20 Sep 2010 19:49:43 -0700
Subject: [PATCH 2/2] Add Exherbo welcome message

---
 src/util.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/util.c b/src/util.c
index f762f97..2ad89b7 100644
--- a/src/util.c
+++ b/src/util.c
@@ -3050,6 +3050,8 @@ void status_welcome(void) {
         free(codename);
 #elif defined(TARGET_ARCH)
         status_printf("Welcome to \x1B[1;36mArch Linux\x1B[0m!\n"); /* Cyan for Arch */
+#elif defined(TARGET_EXHERBO)
+        status_printf("Welcome to \x1B[0;32mExherbo Linux\x1B[0m!\n"); /* Green for Exherbo */
 #else
 #warning "You probably should add a welcome text logic here."
 #endif
-- 
1.7.3.2

