From 38ff330aaab354ddb85780be55b9d7a63f123ae9 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Mon, 20 Sep 2010 19:49:43 -0700
Subject: [PATCH 3/3] Add Exherbo welcome message

---
 src/util.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/util.c b/src/util.c
index bdc194e..26afa71 100644
--- a/src/util.c
+++ b/src/util.c
@@ -3011,6 +3011,8 @@ void status_welcome(void) {
         free(r);
 #elif defined(TARGET_ARCH)
         status_printf("Welcome to \x1B[1;36mArch Linux\x1B[0m!\n"); /* Cyan for Arch */
+#elif defined(TARGET_EXHERBO)
+        status_printf("Welcome to \x1B[0;32mExherbo Linux\x1B[0m!\n"); /* Green for Exherbo */
 #else
 #warning "You probably should add a welcome text logic here."
 #endif
-- 
1.7.2.3

