From 9efaae93df0230d134a2659872f86bef937d4781 Mon Sep 17 00:00:00 2001
From: Dan Callaghan <djc@djc.id.au>
Date: Sun, 3 Jul 2011 17:08:11 +1000
Subject: [PATCH] obey configured prefix for tmpfiles.d and binfmt.d
Upstream: https://bugs.freedesktop.org/show_bug.cgi?id=38686


diff --git a/Makefile.am b/Makefile.am
index 954c3b7..8a4b9ba 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -52,6 +52,7 @@ bashcompletiondir=$(sysconfdir)/bash_completion.d
 pkgsysconfdir=$(sysconfdir)/systemd
 userunitdir=$(prefix)/lib/systemd/user
 tmpfilesdir=$(prefix)/lib/tmpfiles.d
+binfmtdir=$(prefix)/lib/binfmt.d
 usergeneratordir=$(pkglibexecdir)/user-generators
 pkgincludedir=$(includedir)/systemd
 
@@ -94,6 +95,8 @@ AM_CPPFLAGS = \
 	-DRANDOM_SEED=\"$(localstatedir)/lib/random-seed\" \
 	-DSYSTEMD_CRYPTSETUP_PATH=\"$(rootlibexecdir)/systemd-cryptsetup\" \
 	-DSYSTEM_GENERATOR_PATH=\"$(systemgeneratordir)\" \
+	-DSITE_TMPFILES_DIR=\"$(tmpfilesdir)\" \
+	-DSITE_BINFMT_DIR=\"$(binfmtdir)\" \
 	-DUSER_GENERATOR_PATH=\"$(usergeneratordir)\" \
 	-DSYSTEM_SHUTDOWN_PATH=\"$(systemshutdowndir)\" \
 	-DSYSTEMD_KBD_MODEL_MAP=\"$(pkgdatadir)/kbd-model-map\" \
@@ -1403,7 +1406,7 @@ nodist_systemunit_DATA += \
 
 binfmt-install-data-hook:
 	$(MKDIR_P) -m 0755 \
-		$(DESTDIR)$(prefix)/lib/binfmt.d \
+		$(DESTDIR)/lib/binfmt.d \
 		$(DESTDIR)$(sysconfdir)/binfmt.d \
 		$(DESTDIR)$(systemunitdir)/sysinit.target.wants
 	( cd $(DESTDIR)$(systemunitdir)/sysinit.target.wants && \
diff --git a/src/binfmt/binfmt.c b/src/binfmt/binfmt.c
index 954c3b7..8a4b9ba 100644
--- a/src/binfmt/binfmt.c
+++ b/src/binfmt/binfmt.c
@@ -141,7 +141,7 @@ int main(int argc, char *argv[]) {
                                     "/run/binfmt.d",
                                     "/etc/binfmt.d",
                                     "/usr/local/lib/binfmt.d",
-                                    "/usr/lib/binfmt.d",
+                                    SITE_BINFMT_DIR,
                                     NULL);
 
                 if (r < 0) {
diff --git a/src/tmpfiles.c b/src/tmpfiles.c
index 954c3b7..8a4b9ba 100644
--- a/src/tmpfiles.c
+++ b/src/tmpfiles.c
@@ -1095,7 +1095,7 @@ int main(int argc, char *argv[]) {
                                     "/run/tmpfiles.d",
                                     "/etc/tmpfiles.d",
                                     "/usr/local/lib/tmpfiles.d",
-                                    "/usr/lib/tmpfiles.d",
+                                    SITE_TMPFILES_DIR,
                                     NULL);
                 if (r < 0) {
                         r = EXIT_FAILURE;
-- 
1.7.5.4

