From 3a7a1820d4cecbd77c7b74c785af5942510bf080 Mon Sep 17 00:00:00 2001
From: Sergey Poznyakoff <gray@gnu.org.ua>
Date: Thu, 22 Jul 2010 10:13:34 +0000
Subject: Minor fix.

* src/filetypes.h: Remove declarations of stat and lstat.
---
diff --git a/src/filetypes.h b/src/filetypes.h
index f80faab..81f0c32 100644
--- a/src/filetypes.h
+++ b/src/filetypes.h
@@ -81,5 +81,3 @@
 #ifndef S_ISLNK
 #define lstat stat
 #endif
-int lstat ();
-int stat ();
--
cgit v0.9.0.2
