Source: Upstream
Upstream: Duh.
Reason: Fix compilation.

--- podofo/trunk/examples/pdfcontentsgraph/PdfContentsGraph.h	2011/03/21 21:08:05	1443
+++ podofo/trunk/examples/pdfcontentsgraph/PdfContentsGraph.h	2011/03/21 21:31:42	1444
@@ -1,12 +1,10 @@
 #ifndef _PODOFO_PDFCONTENTSGRAPH_H
 #define _PODOFO_PDFCONTENTSGRAPH_H
 
-#include "PdfDefines.h"
+#include "podofo/podofo.h"
 
 #if defined(PODOFO_HAVE_BOOST)
 
-#include "PdfVariant.h"
-
 #include <utility>
 #include <string>
 
