From e0eb742089e42104bfa6a80d7f5f862b211c2416 Mon Sep 17 00:00:00 2001
From: Wulf C. Krueger <philantrop@exherbo.org>
Date: Sat, 1 May 2010 14:43:57 +0200
Subject: [PATCH] Make lua properly optional.
Upstream: Yes.

---
 CMakeLists.txt |   25 ++++++++++++++-----------
 1 files changed, 14 insertions(+), 11 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7e20f65..2270c35 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -395,17 +395,20 @@ ELSE(DEFINED JPEG_RUNTIME_COMPATIBLE)
 	ENDIF(NOT WIN32)
 ENDIF(DEFINED JPEG_RUNTIME_COMPATIBLE)
 
-FIND_PACKAGE(LUA)
-IF(LUA_FOUND)
-	# If we have lua, we can build podofoimpose.
-	MESSAGE("Lua found - PoDoFoImpose will be built with Lua support")
-	MESSAGE(" * Lua include directory: ${LUA_INCLUDE_DIR}")
-	MESSAGE(" * Lua libraries: ${LUA_LIBRARIES}")
-	INCLUDE_DIRECTORIES(${LUA_INCLUDE_DIR})
-	SET(PODOFO_HAVE_LUA TRUE)
-ELSE(LUA_FOUND)
-	MESSAGE("Lua not found - PoDoFoImpose will be built without Lua support")
-ENDIF(LUA_FOUND)
+OPTION(WITH_LUA "Enable Lua support for PoDoFoImpose" ON)
+if(WITH_LUA)
+	FIND_PACKAGE(LUA REQUIRED)
+	IF(LUA_FOUND)
+		# If we have lua, we can build podofoimpose.
+		MESSAGE("Lua found - PoDoFoImpose will be built with Lua support")
+		MESSAGE(" * Lua include directory: ${LUA_INCLUDE_DIR}")
+		MESSAGE(" * Lua libraries: ${LUA_LIBRARIES}")
+		INCLUDE_DIRECTORIES(${LUA_INCLUDE_DIR})
+		ADD_DEFINITIONS(-DPODOFO_HAVE_LUA)
+	ELSE(LUA_FOUND)
+		MESSAGE("Lua not found - PoDoFoImpose will be built without Lua support")
+	ENDIF(LUA_FOUND)
+ENDIF(WITH_LUA)
 
 
 # Check if we should build a multithreaded version of PODOFO
-- 
1.7.1

