Source: upstream's svn
Reason: fix building with gcc-4.7

Index: test/unit/TestUtils.cpp
===================================================================
--- test/unit/TestUtils.cpp	(revision 1492)
+++ test/unit/TestUtils.cpp	(revision 1493)
@@ -17,12 +17,12 @@
  *   Free Software Foundation, Inc.,                                       *
  *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
  ***************************************************************************/
-
 #include "TestUtils.h"
 
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 
 #if defined(_WIN32) || defined(_WIN64)
 #include <windows.h>
