Source: Upstream
Upstream: Duh.
Reason: Fix compilation.

--- podofo/trunk/examples/pdfcontentsgraph/PdfContentsGraph.cpp	2011/03/21 21:08:05	1443
+++ podofo/trunk/examples/pdfcontentsgraph/PdfContentsGraph.cpp	2011/03/21 21:31:42	1444
@@ -4,11 +4,6 @@
 #error This module requires boost::graph
 #endif
 
-#include "PdfContentsTokenizer.h"
-#include "PdfOutputDevice.h"
-#include "PdfOutputStream.h"
-#include "PdfInputStream.h"
-
 #include <cassert>
 #include <string>
 #include <iostream>
