From c8c4a5b19cc2bac816c97bbf830f2a2c0f4f30ad Mon Sep 17 00:00:00 2001
From: Wulf C. Krueger <philantrop@exherbo.org>
Date: Tue, 22 Mar 2011 22:17:21 +0100
Subject: [PATCH] Make LUA optional.

---
 CMakeLists.txt |   24 +++++++++++++-----------
 1 files changed, 13 insertions(+), 11 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b23e6b0..6f81c7c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -407,17 +407,19 @@ ELSE(DEFINED JPEG_RUNTIME_COMPATIBLE)
 	ENDIF(NOT WIN32)
 ENDIF(DEFINED JPEG_RUNTIME_COMPATIBLE)
 
-FIND_PACKAGE(LUA)
-IF(LUA_FOUND)
-	# If we have lua, we can build podofoimpose.
-	MESSAGE("Lua found - PoDoFoImpose and PoDoFoColor will be built with Lua support")
-	MESSAGE(" * Lua include directory: ${LUA_INCLUDE_DIR}")
-	MESSAGE(" * Lua libraries: ${LUA_LIBRARIES}")
-	INCLUDE_DIRECTORIES(${LUA_INCLUDE_DIR})
-	SET(PODOFO_HAVE_LUA TRUE)
-ELSE(LUA_FOUND)
-	MESSAGE("Lua not found - PoDoFoImpose and PoDoFoColor will be built without Lua support")
-ENDIF(LUA_FOUND)
+IF(WANT_LUA)
+    FIND_PACKAGE(LUA)
+    IF(LUA_FOUND)
+            # If we have lua, we can build podofoimpose.
+	    MESSAGE("Lua found - PoDoFoImpose and PoDoFoColor will be built with Lua support")
+	    MESSAGE(" * Lua include directory: ${LUA_INCLUDE_DIR}")
+	    MESSAGE(" * Lua libraries: ${LUA_LIBRARIES}")
+	    INCLUDE_DIRECTORIES(${LUA_INCLUDE_DIR})
+	    SET(PODOFO_HAVE_LUA TRUE)
+    ELSE(LUA_FOUND)
+	    MESSAGE("Lua not found - PoDoFoImpose and PoDoFoColor will be built without Lua support")
+    ENDIF(LUA_FOUND)
+ENDIF(WANT_LUA)
 
 
 # Check if we should build a multithreaded version of PoDoFo
-- 
1.7.4.1

