diff -Naur suck-4.3.2.orig/Makefile.in suck-4.3.2/Makefile.in
--- suck-4.3.2.orig/Makefile.in	2008-09-30 18:30:32.000000000 +0200
+++ suck-4.3.2/Makefile.in	2008-09-30 18:31:04.000000000 +0200
@@ -108,7 +108,7 @@
 PKGNAME=suck
 VERSION_MAJOR=4
 VERSION_MINOR=3
-VERSION_PATCH=2.1
+VERSION_PATCH=2.@PATCHLEVEL@
 VERSION=$(VERSION_MAJOR).$(VERSION_MINOR).$(VERSION_PATCH)
 
 CC = @CC@
@@ -146,7 +146,7 @@
             both_phrases.o lmove_phrases.o
 lmoveobjs = both.o both_phrases.o lmove.o lmove_phrases.o
  
-all: phrases.h suck rpost testhost lmove
+all: phrases.h suck rpost testhost lmove lpost readdb
 	@echo make done
 
 suck: phrases.h $(suckobjs)
@@ -181,20 +181,25 @@
 #install_Spanish: install_bin install_sman
 
 install_bin: suck rpost testhost lmove
-#	- mkdir -p $(bindir)
+	- mkdir -p ${DESTDIR}$(bindir)
 #	- strip -s suck rpost testhost lmove
-	$(INSTALL_PROGRAM) suck $(bindir)/suck
-	$(INSTALL_PROGRAM) rpost $(bindir)/rpost
-	$(INSTALL_PROGRAM) testhost $(bindir)/testhost
-	$(INSTALL_PROGRAM) lmove $(bindir)/lmove
-
-install_man: $(MAN)/suck.1 $(MAN)/rpost.1 $(MAN)/testhost.1 $(MAN)/lmove.1
-	- mkdir -p $(mandir)
-	- rm -f $(mandir)/lpost.$(manext)
-	$(INSTALL_DATA) $(MAN)/suck.1 $(mandir)/suck.$(manext)
-	$(INSTALL_DATA) $(MAN)/rpost.1 $(mandir)/rpost.$(manext)
-	$(INSTALL_DATA) $(MAN)/testhost.1 $(mandir)/testhost.$(manext)
-	$(INSTALL_DATA) $(MAN)/lmove.1 $(mandir)/lmove.$(manext)
+	$(INSTALL_PROGRAM) suck ${DESTDIR}$(bindir)/suck
+	$(INSTALL_PROGRAM) rpost ${DESTDIR}$(bindir)/rpost
+	$(INSTALL_PROGRAM) testhost ${DESTDIR}$(bindir)/testhost
+	$(INSTALL_PROGRAM) lmove ${DESTDIR}$(bindir)/lmove
+	$(INSTALL_PROGRAM) lpost ${DESTDIR}$(bindir)/lpost
+	$(INSTALL_PROGRAM) read_db ${DESTDIR}$(bindir)/read_db
+
+install_man: $(MAN)/suck.1 $(MAN)/rpost.1 $(MAN)/testhost.1 $(MAN)/lmove.1 $(MAN)/lpost.1
+	- mkdir -p ${DESTDIR}$(mandir)
+#	- rm -f $(mandir)/lpost.$(manext)
+	$(INSTALL_DATA) $(MAN)/suck.1 ${DESTDIR}$(mandir)/suck.$(manext)
+	$(INSTALL_DATA) $(MAN)/rpost.1 ${DESTDIR}$(mandir)/rpost.$(manext)
+	$(INSTALL_DATA) $(MAN)/testhost.1 ${DESTDIR}$(mandir)/testhost.$(manext)
+	$(INSTALL_DATA) $(MAN)/lmove.1 ${DESTDIR}$(mandir)/lmove.$(manext)
+	$(INSTALL_DATA) $(MAN)/lpost.1 ${DESTDIR}$(mandir)/lpost.$(manext)
+
+
 
 #install_sman: $(SMAN)/suck.1 $(SMAN)/rpost.1 $(SMAN)/testhost.1 $(SMAN)/lpost.1
 #	$(MAKE) -C $(SMAN)
