--- libcap/include/sys/capability.h.orig	2009-05-13 21:29:29.272302609 +0000
+++ libcap/include/sys/capability.h	2009-05-13 21:28:52.464539435 +0000
@@ -26,9 +26,11 @@
  * Make sure we can be included from userland by preventing
  * capability.h from including other kernel headers
  */
+/*
 #define _LINUX_TYPES_H
 #define _LINUX_FS_H
 #define __LINUX_COMPILER_H
+*/
 #define __user
 
 typedef unsigned int __u32;
