From 3c39b50c48d31aa224c36c06e23d162a564d6496 Mon Sep 17 00:00:00 2001
From: Ingmar Vanhassel <ingmar@exherbo.org>
Date: Sun, 7 Sep 2008 11:56:41 +0200
Subject: [PATCH] Add proper libdir support.

---
 Configure |   27 ++++++++++++++-------------
 1 files changed, 14 insertions(+), 13 deletions(-)

diff --git a/Configure b/Configure
index 77c8ee2..a34d51a 100755
--- a/Configure
+++ b/Configure
@@ -1273,12 +1273,12 @@ libnames=''
 : change the next line if compiling for Xenix/286 on Xenix/386
 xlibpth='/usr/lib/386 /lib/386'
 : Possible local library directories to search.
-loclibpth="/usr/local/lib /opt/local/lib /usr/gnu/lib"
-loclibpth="$loclibpth /opt/gnu/lib /usr/GNU/lib /opt/GNU/lib"
+loclibpth="/usr/local/$exherbo_libdir /opt/local/$exherbo_libdir /usr/gnu/$exherbo_libdir"
+loclibpth="$loclibpth /opt/gnu/$exherbo_libdir /usr/GNU/$exherbo_libdir /opt/GNU/$exherbo_libdir"
 
 : general looking path for locating libraries
-glibpth="/lib /usr/lib $xlibpth"
-glibpth="$glibpth /usr/ccs/lib /usr/ucblib /usr/local/lib"
+glibpth="/$exherbo_libdir /usr/$exherbo_libdir $xlibpth"
+glibpth="$glibpth /usr/ccs/$exherbo_libdir /usr/ucblib /usr/local/$exherbo_libdir"
 test -f /usr/shlib/libc.so && glibpth="/usr/shlib $glibpth"
 test -f /shlib/libc.so     && glibpth="/shlib $glibpth"
 test -d /usr/lib64         && glibpth="$glibpth /lib64 /usr/lib64 /usr/local/lib64"
@@ -6106,8 +6106,8 @@ esac'
 : Reproduce behavior of 5.005 and earlier, maybe drop that in 5.7.
 case "$installstyle" in
 '')	case "$prefix" in
-		*perl*) dflt='lib';;
-		*) dflt='lib/perl5' ;;
+		*perl*) dflt="$exherbo_libdir";;
+		*) dflt="$exherbo_libdir/perl5" ;;
 	esac
 	;;
 *)	dflt="$installstyle" ;;
@@ -6173,8 +6173,8 @@ esac
 : /opt/perl/lib/perl5... would be redundant.
 : The default "style" setting is made in installstyle.U
 case "$installstyle" in
-*lib/perl5*) set dflt privlib lib/$package/$version ;;
-*)	 set dflt privlib lib/$version ;;
+*$exherbo_libdir/perl5*) set dflt privlib $exherbo_libdir/$package/$version ;;
+*)	 set dflt privlib $exherbo_libdir/$version ;;
 esac
 eval $prefixit
 $cat <<EOM
@@ -6405,8 +6405,8 @@ siteprefixexp="$ansexp"
 prog=`echo $package | $sed 's/-*[0-9.]*$//'`
 case "$sitelib" in
 '') case "$installstyle" in
-	*lib/perl5*) dflt=$siteprefix/lib/$package/site_$prog/$version ;;
-	*)	 dflt=$siteprefix/lib/site_$prog/$version ;;
+	*$exherbo_libdir/perl5*) dflt=$siteprefix/$exherbo_libdir/$package/site_$prog/$version ;;
+	*)	 dflt=$siteprefix/$exherbo_libdir/site_$prog/$version ;;
 	esac
 	;;
 *)	dflt="$sitelib"
@@ -10970,9 +10970,9 @@ else
 		else
 			for net in net socket
 			do
-				if test -f /usr/lib/lib$net$_a; then
-					( ($nm $nm_opt /usr/lib/lib$net$_a | eval $nm_extract) ||  \
-					$ar t /usr/lib/lib$net$_a) 2>/dev/null >> libc.list
+				if test -f /usr/$exherbo_libdir/lib$net$_a; then
+					( ($nm $nm_opt /usr/$exherbo_libdir/lib$net$_a | eval $nm_extract) ||  \
+					$ar t /usr/$exherbo_libdir/lib$net$_a) 2>/dev/null >> libc.list
 					if $contains socket libc.list >/dev/null 2>&1; then
 						d_socket="$define"
 						socketlib="-l$net"
@@ -22019,6 +22019,7 @@ eagain='$eagain'
 ebcdic='$ebcdic'
 echo='$echo'
 egrep='$egrep'
+exherbo_libdir='$exherbo_libdir'
 emacs='$emacs'
 endgrent_r_proto='$endgrent_r_proto'
 endhostent_r_proto='$endhostent_r_proto'
-- 
1.6.0.1

