Author: Saleem Abdulrasool <compnerd@compnerd.org>
Reason: Support multiarch on exherbo
Upstream: no
-- 
--- gcc-4.7.2/gcc/config.gcc	2012-12-24 00:00:00.000000000 -0800
+++ gcc-4.7.2/gcc/config.gcc	2012-12-24 08:00:00.000000000 -0800
@@ -239,7 +239,7 @@
 target_gtfiles=
 need_64bit_hwint=
 need_64bit_isa=
-native_system_header_dir=/usr/include
+native_system_header_dir=/usr/${target}/include
 target_type_format_char='@'
 
 # Don't carry these over build->host->target.  Please.
--- gcc-4.7.2/gcc/config/arm/linux-eabi.h	2012-12-24 00:00:00.000000000 -0800
+++ gcc-4.7.2/gcc/config/arm/linux-eabi.h	2012-12-24 00:00:00.000000000 -0800
@@ -103,3 +103,14 @@
 #define CLEAR_INSN_CACHE(BEG, END) not_used
 
 #define ARM_TARGET2_DWARF_FORMAT (DW_EH_PE_pcrel | DW_EH_PE_indirect)
+
+/* exherbo specific: lib paths */
+#undef STANDARD_STARTFILE_PREFIX_1
+#define STANDARD_STARTFILE_PREFIX_1 "/arm-exherbo-linux-gnueabi/lib/"
+
+#undef STANDARD_STARTFILE_PREFIX_2
+#define STANDARD_STARTFILE_PREFIX_2 "/usr/arm-exherbo-linux-gnueabi/lib/"
+
+#undef  GLIBC_DYNAMIC_LINKER
+#define GLIBC_DYNAMIC_LINKER "/usr/arm-exherbo-linux-gnueabi/lib/ld-linux.so.3"
+
--- gcc-4.7.2/gcc/config/rs6000/linux64.h	2012-12-24 00:00:00.000000000 -0800
+++ gcc-4.7.2/gcc/config/rs6000/linux64.h	2012-12-24 00:00:00.000000000 -0800
@@ -548,3 +548,17 @@
 
 /* The default value isn't sufficient in 64-bit mode.  */
 #define STACK_CHECK_PROTECT (TARGET_64BIT ? 16 * 1024 : 12 * 1024)
+
+/* exherbo specific: lib paths */
+#undef STANDARD_STARTFILE_PREFIX_1
+#define STANDARD_STARTFILE_PREFIX_1 "/powerpc64-exherbo-linux-gnueabi/lib/"
+
+#undef STANDARD_STARTFILE_PREFIX_2
+#define STANDARD_STARTFILE_PREFIX_2 "/usr/powerpc64-exherbo-linux-gnueabi/lib/"
+
+#undef GLIBC_DYNAMIC_LINKER32
+#define GLIBC_DYNAMIC_LINKER32 "/usr/powerpc-exherbo-linux-gnueabi/lib/ld.so.1"
+
+#undef GLIBC_DYNAMIC_LINKER64
+#define GLIBC_DYNAMIC_LINKER64 "/usr/powerpc64-exherbo-linux-gnueabi/lib/ld64.so.1"
+
--- gcc-4.7.2/gcc/config/i386/linux.h	2012-12-24 00:00:00.000000000 -0800
+++ gcc-4.7.2/gcc/config/i386/linux.h	2012-12-24 00:00:00.000000000 -0800
@@ -22,3 +22,14 @@
 
 #define GNU_USER_LINK_EMULATION "elf_i386"
 #define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+
+/* exherbo specific: lib paths */
+#undef STANDARD_STARTFILE_PREFIX_1
+#define STANDARD_STARTFILE_PREFIX_1 "/i686-pc-linux-gnu/lib/"
+
+#undef STANDARD_STARTFILE_PREFIX_2
+#define STANDARD_STARTFILE_PREFIX_2 "/usr/i686-pc-linux-gnu/lib/"
+
+#undef GLIBC_DYNAMIC_LINKER
+#define GLIBC_DYNAMIC_LINKER "/usr/i686-pc-linux-gnu/lib/ld-linux.so.2"
+
--- gcc-4.7.2/gcc/config/i386/linux64.h	2012-12-24 00:00:00.000000000 -0800
+++ gcc-4.7.2/gcc/config/i386/linux64.h	2012-12-24 00:00:00.000000000 -0800
@@ -31,3 +31,20 @@
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
 #define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux-x86-64.so.2"
 #define GLIBC_DYNAMIC_LINKERX32 "/libx32/ld-linux-x32.so.2"
+
+/* exherbo specific: lib paths */
+#undef STANDARD_STARTFILE_PREFIX_1
+#define STANDARD_STARTFILE_PREFIX_1 "/x86_64-pc-linux-gnu/lib/"
+
+#undef STANDARD_STARTFILE_PREFIX_2
+#define STANDARD_STARTFILE_PREFIX_2 "/usr/x86_64-pc-linux-gnu/lib/"
+
+#undef GLIBC_DYNAMIC_LINKER32
+#define GLIBC_DYNAMIC_LINKER32 "/usr/x86_64-pc-linux-gnu/ld-linux.so.2
+
+#undef GLIBC_DYNAMIC_LINKER64
+#define GLIBC_DYNAMIC_LINKER64 "/usr/x86_64-pc-linux-gnu/ld-linux-x86-64.so.2"
+
+#undef GLIBC_DYNAMIC_LINKERX32
+#define GLIBC_DYNAMIC_LINKERX32 "/usr/x86_64-pc-linux-gnu/ld-linux-x32.so.2"
+
