Author: Saleem Abdulrasool <compnerd@compnerd.org>
Reason: fix serial configure for dependencies
Upstream: no
-- 
--- gcc-4.8-20130303/configure	2013-03-05 00:00:00.000000000 -0800
+++ gcc-4.8-20130303/configure	2013-03-05 00:00:00.000000000 -0800
@@ -2770,8 +2770,8 @@
 		target-newlib \
 		target-libgomp \
 		target-libatomic \
-		target-libitm \
 		target-libstdc++-v3 \
+		target-libitm \
 		target-libmudflap \
 		target-libsanitizer \
 		target-libssp \
